packages/stateManager/mod.ts

```diff
@@ -213,40 +229,43 @@ class ListingManager extends PublicObjectManager<Listing> {
     }
   }
 
-  async create(l: Partial<Listing>, decimals?: number) {
-    const eventId = await this.client.listing({
-      id: { raw: objectId() },
-      price: { raw: priceToUint256(l.price!, decimals) },
-      metadata: l.metadata,
-      viewState: l.viewState,
+  create(l: Partial<Listing>, decimals?: number) {
+    return this.queueClientRequest(async () => {
+      const eventId = await this.client.listing({
+        id: { raw: objectId() },
+        price: { raw: priceToUint256(l.price!, decimals) },
+        metadata: l.metadata,
+        viewState: l.viewState,
+      });
+
+      // Set up event listener and wait for response after network call
+      return eventListenAndResolve<Listing>(eventId, this, "create");
     });
-    // resolves after the `listing` event has been fired in _processEvent, which happens after the relay accepts the update and has written to the database.
-    return eventListenAndResolve<Listing>(eventId, this, "create");
   }
-
   //update argument passed here will only contain the fields to update.
-  async update(update: Partial<Listing>, decimals?: number) {
-    //ui object to be passed to the network with converted network data types.
-    //we are declaring the update object as type schema.IUpdateListing since we are changing values from hex to bytes and is no longer a interface l
-    const ui: schema.IUpdateListing = {
-      id: { raw: hexToBytes(update.id!) },
-    };
-    if (update.price) {
-      ui.price = { raw: priceToUint256(update.price, decimals) };
-    }
-    if (update.metadata) {
-      ui.metadata = update.metadata;
-    }
-    if (update.viewState !== undefined) {
-      assert(
-        Object.values(ListingViewState).includes(update.viewState),
-        `update.viewState ${update.viewState} must be a valid ListingViewState`,
-      );
-      ui.viewState = update.viewState;
-    }
-    const requestId = await this.client.updateListing(ui);
-    // resolves after the `updateListing` event has been fired, which happens after the relay accepts the update and has written to the database.
-    return eventListenAndResolve<Listing>(requestId, this, "update");
+  update(update: Partial<Listing>, decimals?: number) {
+    return this.queueClientRequest(async () => {
+      const ui: schema.IUpdateListing = {
+        id: { raw: hexToBytes(update.id!) },
+      };
+
+      if (update.price) {
+        ui.price = { raw: priceToUint256(update.price, decimals) };
+      }
+      if (update.metadata) {
+        ui.metadata = update.metadata;
+      }
+      if (update.viewState !== undefined) {
+        assert(
+          Object.values(ListingViewState).includes(update.viewState),
+          `update.viewState ${update.viewState} must be a valid ListingViewState`,
+        );
+        ui.viewState = update.viewState;
+      }
+
+      const requestId = await this.client.updateListing(ui);
+      return eventListenAndResolve<Listing>(requestId, this, "update");
+    });
```

Comment:
I would keep the validation logic out of the queue, since there is no async work going on in it.
```suggestion
      const ui: schema.IUpdateListing = {
        id: { raw: hexToBytes(update.id!) },
      };

      if (update.price) {
        ui.price = { raw: priceToUint256(update.price, decimals) };
      }
      if (update.metadata) {
        ui.metadata = update.metadata;
      }
      if (update.viewState !== undefined) {
        assert(
          Object.values(ListingViewState).includes(update.viewState),
          `update.viewState ${update.viewState} must be a valid ListingViewState`,
        );
        ui.viewState = update.viewState;
      }

    return this.queueClientRequest(async () => {
      const requestId = await this.client.updateListing(ui);
      return eventListenAndResolve<Listing>(requestId, this, "update");
    });
```